### ISSUE TITLE
error: undefined symbol: AER::QV::Avx AER::QV::apply_matrix_avx<double> ...

### ISSUE THREAD

**Issue by yurivict** [Qiskit/qiskit-aer#1525]
### Informations

- **Qiskit Aer version**: 0.10.4
- **Python version**: 3.8
- **Operating system**: FreeBSD 13.1
-  **Compiler**: clang-13

### What is the current behavior?
Build fails:
```
ld: error: undefined symbol: AER::QV::Avx AER::QV::apply_matrix_avx<double>(double*, unsigned long, unsigned long const*, unsigned long, double const*, unsigned long)
>>> referenced by qasm_simulator.cpp
>>>               CMakeFiles/qasm_simulator.dir/contrib/standalone/qasm_simulator.cpp.o:(AER::QV::TransformerAVX2<std::__1::complex<double>*, double>::apply_matrix(std::__1::complex<double>*&, unsigned long, int, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, std::__1::vector<std::__1::complex<double>, std::__1::allocator<std::__1::complex<double> > > const&) const)

ld: error: undefined symbol: AER::QV::Avx AER::QV::apply_diagonal_matrix_avx<double>(double*, unsigned long, unsigned long const*, unsigned long, double const*, unsigned long)
>>> referenced by qasm_simulator.cpp
>>>               CMakeFiles/qasm_simulator.dir/contrib/standalone/qasm_simulator.cpp.o:(AER::QV::TransformerAVX2<std::__1::complex<double>*, double>::apply_diagonal_matrix(std::__1::complex<double>*&, unsigned long, int, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, std::__1::vector<std::__1::complex<double>, std::__1::allocator<std::__1::complex<double> > > const&) const)

ld: error: undefined symbol: AER::QV::Avx AER::QV::apply_matrix_avx<float>(float*, unsigned long, unsigned long const*, unsigned long, float const*, unsigned long)
>>> referenced by qasm_simulator.cpp
>>>               CMakeFiles/qasm_simulator.dir/contrib/standalone/qasm_simulator.cpp.o:(AER::QV::TransformerAVX2<std::__1::complex<float>*, float>::apply_matrix(std::__1::complex<float>*&, unsigned long, int, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, std::__1::vector<std::__1::complex<double>, std::__1::allocator<std::__1::complex<double> > > const&) const)

ld: error: undefined symbol: AER::QV::Avx AER::QV::apply_diagonal_matrix_avx<float>(float*, unsigned long, unsigned long const*, unsigned long, float const*, unsigned long)
>>> referenced by qasm_simulator.cpp
>>>               CMakeFiles/qasm_simulator.dir/contrib/standalone/qasm_simulator.cpp.o:(AER::QV::TransformerAVX2<std::__1::complex<float>*, float>::apply_diagonal_matrix(std::__1::complex<float>*&, unsigned long, int, std::__1::vector<unsigned long, std::__1::allocator<unsigned long> > const&, std::__1::vector<std::__1::complex<double>, std::__1::allocator<std::__1::complex<double> > > const&) const)
```

### Steps to reproduce the problem
Regular build.

### What is the expected behavior?
successful build

### Suggested solutions
n/a

**Comment by hhorii** [comment]
We do not test Aer on FreeBSD in our CI. This problem may from OS-dependent issue.

**Comment by yurivict** [comment]
@hhorii 
> This problem may from OS-dependent issue.

This is unlikely.
Do you test with clang?

**Comment by yurivict** [comment]
This PR solves this problem: https://github.com/Qiskit/qiskit-aer/pull/1576

**Comment by hhorii** [comment]
Thanks. I'm using apple silicon...

**Comment by jakelishman** [comment]
Fixed by #1576.  (I'm assuming - please re-open if not.)
